File: experiment_jp_test/j2pyrawpy/for__0__MAXIMUM_SUM_INCREASING_SUBSEQUENCE_FROM_A_PREFIX_AND_A_GIVEN_ELEMENT_AFTER_PREFIX_IS_MUST.py
#!/usr/bin/env python
""" generated source for module for__0__MAXIMUM_SUM_INCREASING_SUBSEQUENCE_FROM_A_PREFIX_AND_A_GIVEN_ELEMENT_AFTER_PREFIX_IS_MUST """
class X(object):
    """ generated source for class X """
    @classmethod
    def pre_compute(cls, a, n, index, k):
        """ generated source for method pre_compute """
        dp = [None]*n
        i = 1
        while i < n:
            while j < n:
                if a[j] > a[i] and j > i:
                    if dp[i - 1][i] + a[j] > dp[i - 1][j]:
                        dp[i][j] = dp[i - 1][i] + a[j]
                    else:
                        dp[i][j] = dp[i - 1][j]
                else:
                    dp[i][j] = dp[i - 1][j]
                j += 1
            i += 1
        return dp[index][k]

File: experiment_jp_test/mutants/for__0__MAXIMUM_SUM_INCREASING_SUBSEQUENCE_FROM_A_PREFIX_AND_A_GIVEN_ELEMENT_AFTER_PREFIX_IS_MUST.java
static int pre_compute ( int a [ ] , int n , int index , int k ) { int dp [ ] [ ] = new int [ n ] [ n ] ; for ( int i = 1 ; i < n ; i ++ ) { for ( int j = 0 ; j < n ; j ++ ) { if ( a [ j ] > a [ i ] && j > i ) { if ( dp [ i - 1 ] [ i ] + a [ j ] > dp [ i - 1 ] [ j ] ) dp [ i ] [ j ] = dp [ i - 1 ] [ i ] + a [ j ] ; else dp [ i ] [ j ] = dp [ i - 1 ] [ j ] ; } else dp [ i ] [ j ] = dp [ i - 1 ] [ j ] ; } } return dp [ index ] [ k ] ; }
